privilege/privileges/cache.go

```diff
@@ -940,6 +940,18 @@ func (p *MySQLPrivilege) getDefaultRoles(user, host string) []*auth.RoleIdentity
 	return ret
 }
 
+func (p *MySQLPrivilege) getAllRoles(user, host string) []*auth.RoleIdentity {
+	ret := make([]*auth.RoleIdentity, 0)
+	key := user + "@" + host
+	edgeTable, ok := p.RoleGraph[key]
```

Comment:
```
if !ok {
    return nil
}

ret := make([]*auth.RoleIdentity, 0, len(edgeTable.roleList))
```

When you're writing Go, remind:
1. test the branch that could return early
2. write cap for make() whenever possible

Reply:
fixed @tiancaiamao